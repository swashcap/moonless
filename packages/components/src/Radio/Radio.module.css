/** Radio */

/** Radio: Label */

.RadioLabel {
  display: flex;
}

/** Radio: Indicator */

.RadioIndicator {
  background: var(--component-radio-indicator-states-base-background-color-base);
  border-color: var(--component-radio-indicator-states-base-border-color-base);
  border-radius: 100%;
  border-style: solid;
  border-width: var(--component-radio-indicator-border-width);
  cursor: pointer;
  height: var(--component-radio-indicator-height);
  margin-right: var(--component-radio-label-margin-left);
  width: var(--component-radio-indicator-width);
}

.RadioIndicator:hover,
:global(.storybook-hover) .RadioIndicator {
  background: var(--component-radio-indicator-states-base-background-color-hover);
  border-color: var(--component-radio-indicator-states-base-border-color-hover);
}

.RadioIndicator:focus,
:global(.storybook-focus) .RadioIndicator,
.RadioIndicator:active,
:global(.storybook-active) .RadioIndicator {
  background: var(--component-radio-indicator-states-base-background-color-active);
  border-color: var(--component-radio-indicator-states-base-border-color-active);
}

/** Radio: Indicator: Checked */

/* stylelint-disable no-descending-specificity */

input:checked ~ .RadioIndicator {
  background: var(--component-radio-indicator-states-checked-background-color-base);
  border-color: var(--component-radio-indicator-states-checked-border-color-base);
  position: relative;
}

input:checked ~ .RadioIndicator::after {
  background: var(--component-radio-dot-states-checked-background-color);
  border-radius: 100%;
  content: '';
  height: 0.75em;
  left: 0.325em;
  position: absolute;
  top: 0.325em;
  width: 0.75em;
}

input:checked ~ .RadioIndicator:hover,
:global(.storybook-hover) input:checked ~ .RadioIndicator {
  background: var(--component-radio-indicator-states-checked-background-color-hover);
  border-color: var(--component-radio-indicator-states-checked-border-color-hover);
}

input:checked ~ .RadioIndicator:focus,
:global(.storybook-focus) input:checked ~ .RadioIndicator,
input:checked ~ .RadioIndicator:active,
:global(.storybook-active) input:checked ~ .RadioIndicator {
  background: var(--component-radio-indicator-states-checked-background-color-active);
  border-color: var(--component-radio-indicator-states-checked-border-color-active);
}

/** Radio: Indicator: Disabled */

input[disabled] ~ .RadioIndicator {
  background: var(--component-radio-indicator-states-base-background-color-disabled);
  border-color: var(--component-radio-indicator-states-base-border-color-disabled);
  cursor: default;
}

input[disabled]:checked ~ .RadioIndicator {
  background: var(--component-radio-indicator-states-checked-background-color-disabled);
  border-color: var(--component-radio-indicator-states-checked-border-color-disabled);
}

/* stylelint-enable no-descending-specificity */

/** Radio: Error */

.RadioError {
  margin-top: var(--component-radio-error-margin-top);
}
