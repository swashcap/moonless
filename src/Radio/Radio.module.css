/** Radio */

/** Radio: Label */

.RadioLabel {
  display: flex;
}

/** Radio: Indicator */

.RadioIndicator {
  border-color: var(--theme-color-secondary-border);
  border-radius: 100%;
  border-style: solid;
  border-width: var(--theme-size-medium-border-width);
  cursor: pointer;
  height: 1.5em;
  margin-right: var(--space-100);
  width: 1.5em;
}

.RadioIndicator:hover,
:global(.storybook-hover) .RadioIndicator {
  background: var(--theme-color-secondary-hover-background);
  border-color: var(--theme-color-secondary-hover-border);
}

.RadioIndicator:focus,
:global(.storybook-focus) .RadioIndicator,
.RadioIndicator:active,
:global(.storybook-active) .RadioIndicator {
  background: var(--theme-color-secondary-active-background);
  border-color: var(--theme-color-secondary-active-border);
}

/** Radio: Indicator: Checked */

/* stylelint-disable no-descending-specificity */

input:checked ~ .RadioIndicator {
  background: var(--theme-color-primary-background);
  border-color: var(--theme-color-primary-border);
  position: relative;
}

input:checked ~ .RadioIndicator::after {
  background: var(--theme-color-primary-text);
  border-radius: 100%;
  content: '';
  height: 0.75em;
  left: 0.325em;
  position: absolute;
  top: 0.325em;
  width: 0.75em;
}

input:checked ~ .RadioIndicator:hover,
:global(.storybook-hover) input:checked ~ .RadioIndicator {
  background: var(--theme-color-primary-hover-background);
  border-color: var(--theme-color-primary-hover-border);
}

input:checked ~ .RadioIndicator:focus,
:global(.storybook-focus) input:checked ~ .RadioIndicator,
input:checked ~ .RadioIndicator:active,
:global(.storybook-active) input:checked ~ .RadioIndicator {
  background: var(--theme-color-primary-active-background);
  border-color: var(--theme-color-primary-active-border);
}

/** Radio: Indicator: Disabled */

input[disabled] ~ .RadioIndicator {
  background: var(--theme-color-secondary-disabled-background);
  border-color: var(--theme-color-secondary-disabled-border);
  cursor: default;
}

input[disabled]:checked ~ .RadioIndicator {
  background: var(--theme-color-primary-disabled-background);
  border-color: var(--theme-color-primary-disabled-border);
}

/* stylelint-enable no-descending-specificity */

/** Radio: Error */

.RadioError {
  margin-top: var(--space-100);
}
