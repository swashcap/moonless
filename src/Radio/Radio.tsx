import React from 'react';

import styles from './Radio.module.css';
import visuallyHiddenStyles from '../VisuallyHidden/VisuallyHidden.module.css';
import { useId } from '../utils/useId';

export interface RadioProps
  extends React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  > {
  checked?: boolean;
  disabled?: boolean;
  id?: string;
  label?: React.ReactNode;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => any;
}

export const Radio = React.forwardRef<HTMLInputElement, RadioProps>(
  ({ checked, disabled, id: idProp, label, onChange, ...rest }, ref) => {
    const fallbackId = useId('checkbox-');

    const id = idProp ?? fallbackId;

    return (
      <div {...rest}>
        <label className={styles.RadioLabel} htmlFor={id}>
          <input
            checked={checked}
            className={visuallyHiddenStyles.VisuallyHidden}
            disabled={disabled}
            id={id}
            onChange={onChange}
            ref={ref}
            type="radio"
          />
          <span aria-hidden="true" className={styles.RadioIndicator} />
          {label}
        </label>
      </div>
    );
  }
);
