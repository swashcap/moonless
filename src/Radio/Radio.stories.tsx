import React from 'react';

import { Radio } from './Radio';

export default {
  component: Radio,
  title: 'Radio',
};

export const Default = () => <Radio />;
