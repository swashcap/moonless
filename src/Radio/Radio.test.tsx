import React from 'react';
import { render } from '@testing-library/react';

import { Radio } from './Radio';

describe('Radio', () => {
  test('It should render.', () => {
    const { container } = render(<Radio />);

    expect(container).toMatchSnapshot();
  });
});
