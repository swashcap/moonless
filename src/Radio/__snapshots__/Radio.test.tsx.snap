// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio It should render. 1`] = `
<div>
  <div>
    <label
      class="RadioLabel"
      for="checkbox-1"
    >
      <input
        class="VisuallyHidden"
        id="checkbox-1"
        type="radio"
      />
      <span
        aria-hidden="true"
        class="RadioIndicator"
      />
    </label>
  </div>
</div>
`;
